internal/cmd/environment/command_delete.go

```diff
@@ -28,33 +28,60 @@ func (c *command) newDeleteCommand() *cobra.Command {
 }
 
 func (c *command) delete(cmd *cobra.Command, args []string) error {
-	id := args[0]
+	if err := c.confirmDeletion(cmd, args); err != nil {
+		return err
+	}
 
-	environment, err := c.V2Client.GetOrgEnvironment(id)
-	if err != nil {
-		return errors.NewErrorWithSuggestions(err.Error(), "List available environments with `confluent environment list`.")
+	var errs error
+	var deleted []string
+	for _, id := range args {
+		if err := c.V2Client.DeleteOrgEnvironment(id); err != nil {
+			errs = errors.Join(err)
+		} else {
+			deleted = append(deleted, id)
+			if id == c.Context.GetCurrentEnvironment() {
+				c.Context.SetCurrentEnvironment("")
+
+				if err := c.Config.Save(); err != nil {
+					errs = errors.Join(errs, errors.Wrap(err, errors.EnvSwitchErrorMsg))
+				}
+			}
+			c.Context.DeleteEnvironment(id)
+			_ = c.Config.Save()
+		}
 	}
+	deletion.PrintSuccessfulDeletionMsg(deleted, resource.Environment)
 
-	promptMsg := fmt.Sprintf(errors.DeleteResourceConfirmMsg, resource.Environment, id, environment.GetDisplayName())
-	if _, err := form.ConfirmDeletion(cmd, promptMsg, environment.GetDisplayName()); err != nil {
-		return err
+	if errs != nil {
+		return errors.NewErrorWithSuggestions(errs.Error(), fmt.Sprintf(errors.ListResourceSuggestions, resource.Environment, pcmd.FullParentName(cmd)))
 	}
 
-	if err := c.V2Client.DeleteOrgEnvironment(id); err != nil {
+	return nil
+}
+
+func (c *command) confirmDeletion(cmd *cobra.Command, args []string) error {
+	var displayName string
+	describeFunc := func(id string) error {
+		environment, err := c.V2Client.GetOrgEnvironment(id)
+		if err == nil && id == args[0] {
```

Comment:
why is it always args[0]?

Reply:
Currently, some delete operations ask a user to type a string. The new version will still do that, but only if the user is deleting a single resource (otherwise it just asks for a `y/n` response). So I only need to keep one string.

The reason it's always the first is that this check requires that all id's belong to valid resources. So the only way to reach the string typing prompt is if there's only one resource, and it exists, so the 0 index id.